### ISSUE TITLE
Unused private constructor parameters are not reported by no-unused-variable

### ISSUE THREAD

**Issue by chrismbarr** [palantir/tslint#1314]
### Bug Report

- __TSLint version__: `3.11.0`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: `gulp-tslint`

#### TypeScript code being linted

```ts
class Example {
  private foo = "foo"; //this is NEVER used, and it's correctly reported!

  constructor(
    private bar: string, //This IS used, it's ok!
    private unused: string  //<-- This is NEVER used, but not reported anywhere!
  ){
    this.doStuff();
  }

  private doStuff = () => {  //This IS used, it's ok!
    console.log(this.bar);
  };

  private doNothing = () => {  //this is NEVER used, and it's correctly reported!
    let blah = 5; //this is NEVER used, and it's correctly reported!
  };
}
```

#### Actual Behavior

With the `no-unused-variable` rule, it will correctly report unused private class properties/methods, and unused local variables.

#### Expected Behavior
It should also report unused private constructor parameters.

**Comment by adidahiya** [comment]
Thanks for the bug report @chrismbarr. I think the fix should be fairly simple if anyone wants to pick it up -- we just need to walk constructor parameter property declarations and add them to the list of variables to check in `no-unused-variable`.

**Comment by chrismbarr** [comment]
Doh!  it looks like there's a `check-parameters` option that I overlooked and this is already being [taken into account by this test](https://github.com/palantir/tslint/blob/master/test/rules/no-unused-variable/check-parameters/test.ts.lint#L20-L23)!

Correct me if I'm wrong, but if this covers the issue above then this issue can be closed
